apps/builder/app/canvas/features/wrapper-component/wrapper-component.tsx

```diff
@@ -145,7 +145,13 @@ export const WrapperComponentDev = ({
           />
         }
         onChange={(updates) => {
-          onChangeChildren?.({ instanceId: instance.id, updates });
+          store.createTransaction([rootInstanceContainer], (rootInstance) => {
+            const { instancesById } = createInstancesIndex(rootInstance);
```

Comment:
I wonder if you should be creating a completely new index every time you want to mutate children as opposed to using immer on the existing index

Reply:
Existing index is not wrapped with draft so will not work.
This will be optimized when we represent instances as map. Still need to migrate stuff.